File: android/src/main/java/com/ziggeo/tasks/UploadFileTask.java
package com.ziggeo.tasks;


import androidx.annotation.NonNull;
import androidx.annotation.Nullable;

import com.facebook.react.bridge.Promise;

import java.util.HashMap;

/**
 * Created by Alex Bedulin on 06.02.2018.
 */

public class UploadFileTask extends Task {

    private HashMap<String, String> extraArgs;

    public UploadFileTask(@NonNull Promise promise) {
        super(promise);
    }

    @Nullable
    public HashMap<String, String> getExtraArgs() {
        return extraArgs;
    }

    public void setExtraArgs(@Nullable HashMap<String, String> args){
        extraArgs = args;
    }

}

File: android/src/main/java/com/ziggeo/tasks/UploadFileTask.kt
package com.ziggeo.tasks

import com.facebook.react.bridge.Promise
import java.util.*

/**
 * Created by Alex Bedulin on 06.02.2018.
 */
class UploadFileTask(
        promise: Promise,
        var extraArgs: HashMap<String, String?>? = null
) : Task(promise)